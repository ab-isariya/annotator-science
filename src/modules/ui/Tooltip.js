import React from 'react';
import PropTypes from 'prop-types';

import Tippy from '@tippyjs/react';
import {followCursor} from 'tippy.js';

const Tooltip = ({children, content, size, dark, theme, ...otherProps}) => {
  //Should build out a string like 'annotate-small' or 'annotate-dark-large'.
  //NOTE(Rejon): See theme name definitions in index.css.
  const styleTheme = theme || `annotate${dark ? '-dark' : ''}-${size}`;

  //NOTE(Rejon): We use a fragment around the children so Tippy doesn't randomly
  //             complain about children not being in the type it expects. (String or Node)
  return (
    <Tippy
      plugins={[followCursor]}
      content={content}
      theme={styleTheme}
      {...otherProps}>
      <>{children}</>
    </Tippy>
  );
};

Tooltip.defaultProps = {
  size: 'medium'
};

Tooltip.propTypes = {
  //React Node child that triggers the tooltip to show
  children: PropTypes.node.isRequired,
  //Content that renders inside of the tooltip
  content: PropTypes.any.isRequired,
  //Tooltip theme size (large, medium, or small)
  //medium is default
  //NOTE(Rejon): See index.css for theme size definitions
  size: PropTypes.oneOf(['large', 'medium', 'small']),
  //Tooltip dark theme, if this prop is on the tooltip it will use
  //an alternate dark theme defined in index.css
  dark: PropTypes.any,
  //Custom theme name incase we want to use an existing tippy theme instead.
  theme: PropTypes.string
};

export default Tooltip;
