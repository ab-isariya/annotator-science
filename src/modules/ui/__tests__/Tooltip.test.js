import ReactDOM from 'react-dom';
import Tooltip from '../Tooltip';

import renderer from 'react-test-renderer';

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element, node) => {
    return element;
  });
});

describe('<Tooltip/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer
      .create(<Tooltip content={'World'}>Hello</Tooltip>)
      .toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with dark and a set size', () => {
    const comp_dark = renderer
      .create(
        <Tooltip content={'World'} dark>
          Hello
        </Tooltip>
      )
      .toJSON();

    expect(comp_dark).toMatchSnapshot();

    const comp_size = renderer
      .create(
        <Tooltip content={'Size'} size={'small'}>
          Super
        </Tooltip>
      )
      .toJSON();

    expect(comp_size).toMatchSnapshot();
  });
});
