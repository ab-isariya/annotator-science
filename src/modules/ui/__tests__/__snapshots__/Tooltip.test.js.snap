// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip/> Renders with dark and a set size 1`] = `"Hello"`;

exports[`<Tooltip/> Renders with dark and a set size 2`] = `"Super"`;

exports[`<Tooltip/> Renders without crashing 1`] = `"Hello"`;
